### PR TITLE
Surya/core 5692/list invitations

### PR DESCRIPTION
**Pull request by modalduality**
a decent amount of code but mostly just bookkeeping

there's a few issues I'm running into but I believe they are due to DIARKs.

### CODE REVIEW

File: go/teams/list.go [hunk 1]
```diff
@@ -57,10 +57,12 @@ func List(ctx context.Context, g *libkb.GlobalContext, arg keybase1.TeamListArg)
 		return nil, err
 	}
 
+	teamNames := make(map[string]bool)
 	upakLoader := g.GetUPAKLoader()
```
Comment (modalduality) [inline go/teams/list.go]: still confused as to what upak/upak2 is and when we should use each

File: go/teams/list.go [hunk 2]
```diff
@@ -81,13 +83,51 @@ func List(ctx context.Context, g *libkb.GlobalContext, arg keybase1.TeamListArg)
 		}
 	}
 
+	annotatedInvites := make(map[keybase1.TeamInviteID]keybase1.AnnotatedTeamInvite)
+	for teamName := range teamNames {
+		t, err := GetForTeamManagementByStringName(ctx, g, teamName, true)
```
Comment (modalduality) [inline go/teams/list.go]: I'm forcing admin to be true here, otherwise doesn't display any invitations

File: go/teams/storage.go [hunk 3]
```diff
@@ -80,7 +80,7 @@ type DiskStorage struct {
 }
 
 // Increment to invalidate the disk cache.
-const diskStorageVersion = 2
+const diskStorageVersion = 3
```
Comment (modalduality) [inline go/teams/storage.go]: since I'm changing sigstate avdl

File: go/teams/list.go (hunk 2 above)
Comment (patrickxb) [inline go/teams/list.go]: Does it still work if not an admin?  i.e. does it return an error?

**Comment by patrickxb** [comment]
Looks good at first glance, but could you add some tests?  There are some existing tests in teams/member_test.go like MemberAddSocial, MemberAddNoPUK, MemberAddNoKeys, MemberAddEmail that generate invite links.  You could test what you've added here as well.

File: go/teams/list.go (hunk 2 above)
Comment (modalduality) [inline go/teams/list.go]: yeah I was just looking at this, I'll need to decide whether to do this based on server response. However right now the endpoint for --all doesn't return implicit param, so I need to do that first.

**Comment by modalduality** [comment]
@patrickxb I reworked this to print implicit admins more in like with ccsandbox, and tested invitations in member_test.go.

**Comment by keybase-ci-visdiff** [comment]
The commits 6b24ba19756b...2c17ff8545d4 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501070569071-6b24ba19756b-2c17ff8545d4-linux/Markdown-emoji-6b24ba19756b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1501070569071-6b24ba19756b-2c17ff8545d4-linux/Markdown-emoji-2c17ff8545d4.png)  
   Markdown-emoji comparison
</summary>

File: go/teams/list.go [hunk 4]
```diff
@@ -52,21 +52,32 @@ func List(ctx context.Context, g *libkb.GlobalContext, arg keybase1.TeamListArg)
 		uid = res.GetUID()
 	}
 
+	me, err := libkb.LoadMe(libkb.NewLoadUserArg(g))
```
Comment (patrickxb) [inline go/teams/list.go]: You just need this for the current user's UID, correct?  If so,

    g.ActiveDevice.UID()

will be less resource-intensive...

**Pull request merged**